docs(operator): `openshiftRoute` ingress type (#3547)
kroxylicious/kroxylicious: kroxylicious-docs/docs/_modules/configuring/con-kafkaproxyingress-overview.adoc

You must also xref:proc-virtualkafkacluster-loadbalancer-dns-config-{context}[configure DNS] so that the bootstrap and broker address resolve from the network used by the applications.

== `openShiftRoute` ingress type

IMPORTANT: OpenShift Routes are an OpenShift-only feature and are not supported on vanilla Kubernetes environments. The Kroxylicious Operator actively detects whether the OpenShift Route API is supported on the Kubernetes Server; if the feature is not available, the operator will gracefully reject any attempt by the user to use it.

The `openShiftRoute` ingress type allows applications running off-cluster to connect to the virtual cluster. TLS must be used with this ingress type.

The `openShiftRoute` ingress type uses OpenShift `Route` resources to enable off-cluster access.

When using an `openShiftRoute` ingress, the proxy uses SNI (Server Name Indication) to match the client’s requested host name to the correct virtual cluster and broker within the proxy.

> The bootstrap and broker addresses are automatically generated. They look like this:

Reviewer: ```suggestion
The bootstrap and broker addresses are generated automatically, for example:
```
